### ISSUE TITLE
Key incompatibility with setuptools >= 74, fixed in 1.81.2 or 1.82.x

### ISSUE THREAD

**Issue by ehfd** [conda-forge/gobject-introspection-feedstock#77]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Glib and gobject-introspection must be updated ASAP. Otherwise, for lower versions, there should be a pinning of setuptools < 74 for any version lower than 1.81.2 (https://gitlab.gnome.org/GNOME/gobject-introspection/-/tags).
CC @hmaarrfk

https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/515
https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/490

```
Traceback (most recent call last):
  File "$BUILD_PREFIX/bin/g-ir-scanner", line 103, in <module>
    from giscanner.scannermain import scanner_main
  File "$BUILD_PREFIX/lib/gobject-introspection/giscanner/scannermain.py", line 35, in <module>
    from giscanner.ast import Include, Namespace
  File "$BUILD_PREFIX/lib/gobject-introspection/giscanner/ast.py", line 27, in <module>
    from .sourcescanner import CTYPE_TYPEDEF, CSYMBOL_TYPE_TYPEDEF
  File "$BUILD_PREFIX/lib/gobject-introspection/giscanner/sourcescanner.py", line 25, in <module>
    from .ccompiler import CCompiler
  File "$BUILD_PREFIX/lib/gobject-introspection/giscanner/ccompiler.py", line 29, in <module>
    from distutils.msvccompiler import MSVCCompiler
ModuleNotFoundError: No module named 'distutils.msvccompiler'
```

### Installed packages

```shell
Known issue of g-ir-scanner.
```

### Environment info

```shell
Known issue of g-ir-scanner.
```

**Comment by hmaarrfk** [comment]
Thanks for the workaround.

It isn't for lack of trying, https://github.com/conda-forge/gobject-introspection-feedstock/pull/76
but compilation of this package is never easy for me....


Lets try to backport the fix?
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
We don't ship "odd" versions since they are unstable.

But i would backport the fix or take help fixing the new version.

Help appreciated.

**Comment by ehfd** [comment]
Note that 1.82.x has this patch for setuptools 74 so the patch is not necessary.

**Comment by pkgw** [comment]
This should now be solved with the merger of #78. I'll leave this open since I haven't sat down and confirmed that.

**Comment by ehfd** [comment]
This should be fixed, although a new issue https://github.com/conda-forge/glib-feedstock/issues/191 is a build blocker.
